sources/api/pluto/src/proxy.rs

```diff
@@ -33,6 +34,42 @@ pub(crate) fn fetch_proxy_env() -> (Option<String>, Option<String>) {
     (https_proxy, no_proxy)
 }
 
+#[derive(Debug, Eq, PartialEq)]
+struct ProxyInfo {
+    proxy_url: String,
+    username: Option<String>,
+    password: Option<String>,
+}
+
+// Check for presence of proxy auth info to add to header, reformat https_proxy to be proxy URL
+fn parse_proxy_info(https_proxy: &str) -> ProxyInfo {
+    match https_proxy.rsplit_once('@') {
+        Some((p, s)) => {
+            // Check for scheme
+            if let Some((scheme, auth_info)) = p.split_once("://") {
+                let userpass = auth_info.split_once(':');
```

Comment:
Is there a URL parsing crate we should use here instead? What if the username or password contains `:` or `@`? I assume this would be invalid (not URL encoded), but the type (`&str`) doesn't guarantee it.